From f6d021512a3d076c8a559e52ea1d1c84ff5c4154 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 18 May 2015 11:19:14 +0800
Subject: GtkWindow: Don't assume no shadow width even when not supported

Even if a window doesn't support client side shadow
(gtk_window_supports_client_shadow returns FALSE), don't assume the
shadow width is zero, as CSD may have been enabled anyway (meaning
priv->client_decorated is TRUE). In that case we still need to report
the correct width.

https://bugzilla.gnome.org/show_bug.cgi?id=749451

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index 9c147fb..423c6bd 100644
--- gtk/gtkwindow.c
+++ gtk/gtkwindow.c
@@ -6480,9 +6480,12 @@ get_shadow_width (GtkWidget *widget,
 
   *shadow_width = border;
 
-  if (!priv->decorated ||
-      (!gtk_window_should_use_csd (window) && !priv->client_decorated) ||
-      !gtk_window_supports_client_shadow (window))
+  if (!priv->decorated)
+    return;
+
+  if (!priv->client_decorated &&
+      !(gtk_window_should_use_csd (window) &&
+        gtk_window_supports_client_shadow (window)))
     return;
 
   if (priv->maximized ||
-- 
cgit v0.10.2

